### DISCUSSION TITLE
About changing the message "App data is not configured"

### DISCUSSION THREAD

**Discussion by duongdominhchau** [actix/actix-web#2150] [General] [answered]
> App data is not configured, to configure use App::data()

This message doesn't provide useful information to locate the error line. I got this when I declare wrong extractor type in my handler. Is there any other case where this error is displayed? If not, I think we should mention in the error message that this is caused by a failed extraction. If the handler name can also be displayed (if that handler has a name), it is much better.

**Comment by robjtede** [comment, marked as answer]
In the v4 beta this error message is improved to show the type name that could not be extracted.

**Reply by enaut** [reply]
Mhm I'm just porting to 4.0.0-beta.9 and I get `App data is not configured, to configure use App::data()` however I have no clue what goes wrong as there is no precise error and when updating there are so many changes so I cannot really bisect :( .

**Reply by robjtede** [reply]
https://github.com/actix/actix-web/blob/master/src/data.rs#L131 enable debug logging for the improved error
